Changes for satistics window

defnax opened the pull request:
* Added to display avatar icons for global router & gxs transport
* Fixed to work context menu call for details View
* Updated Person details windows
* Display the receive/send time with date format

defnax committed:
Merge pull request #17 from RetroShare/master

Improve error reporting in createChannelV2

defnax committed:
Merge pull request #18 from RetroShare/master

update master

defnax committed:
Merge pull request #19 from RetroShare/master

update to master

defnax committed:
Merge pull request #20 from RetroShare/master

update to master

defnax committed:
Added to display avatar icons & fixes for context menu & format send/receive time

Added to display avatar icons for global router & gxs transport
Fixed to work context menu call for details View
Updated Person details windows
Display the receive/send time with date format

csoler commented on retroshare-gui/src/gui/statistics/GlobalRouterStatistics.cpp:
```diff
@@ -177,6 +180,18 @@ void GlobalRouterStatistics::updateContent()
         
         if(nicknames.isEmpty())
           nicknames = tr("Unknown");
+	  
```
You should directly use a GxsIdTreeWidgetItem: it manages the display of avatar, a proper tooltip, etc!

csoler commented on retroshare-gui/src/gui/statistics/GlobalRouterStatistics.cpp:
```diff
@@ -185,9 +200,9 @@ void GlobalRouterStatistics::updateContent()
-        item -> setData(COL_RECEIVED,     Qt::DisplayRole, QString::number(now - cache_infos[i].routing_time));
-        item -> setData(COL_SEND,         Qt::DisplayRole, QString::number(now - cache_infos[i].last_sent_time));
-        item -> setData(COL_DUPLICATION_FACTOR, Qt::DisplayRole, QString::number(cache_infos[i].duplication_factor));
+		item -> setData(COL_RECEIVED, Qt::DisplayRole, DateTime::formatDateTime(routingtime));
```
For debugging, it's better to display the delta than the actual time, since in the end what you want is the delta.

defnax replied:
i had it first both the thinked remove that with seconds,
i can add it again

defnax committed:
Added back the routing & last sent time in (secs ago)

csoler commented:
There a few small problems:
- Sending time in gxs transport statistics says "secs ago" but it shows a date. It should show seconds instead.
- the tooltip on the Ids in Destination do not show the avatar. You should probably use a GxsIdRsTreeWidgetItem instead.

defnax commented:
> * Sending time in gxs transport statistics says "secs ago" but it shows a date. It should show seconds instead.

but thats not added by me, that was always there not sure who did that, needs a second column for show the seconds

defnax committed:
Fixed to use GxsIdRSTreeWidgetItem on Pending packets & pending data items

Fixed to use GxsIdRSTreeWidgetItem on Pending packets & pending data items
Added to display on Pending data items the time in seconds
Set text alingment for some items

defnax commented:
> There a few small problems:
> 
> * Sending time in gxs transport statistics says "secs ago" but it shows a date. It should show seconds instead.

added new column for that

> * the tooltip on the Ids in Destination do not show the avatar. You should probably use a GxsIdRsTreeWidgetItem instead.

i has used now for pending packets & pending data items tree GxsIdRsTreeWidgetItem
but the other tree on gxs transport groups wasnt able to use i get conflicts there.

when replacing this
> QTreeWidgetItem *sitem ;
> {
> QString unique_id = QString::fromStdString(meta.mMsgId.toStdString());
> 
> QList<QTreeWidgetItem*> iteml = groupTreeWidget->findItems(unique_id,Qt::MatchExactly,COL_GROUP_UNIQUE_ID) ;
> 
> if(iteml.empty())
> 	sitem = new QTreeWidgetItem(item) ;
> 	else
> 	sitem = *iteml.begin();
> }

with this:

> GxsIdRSTreeWidgetItem *item = new 
> GxsIdRSTreeWidgetItem(NULL,GxsIdDetails::ICON_TYPE_AVATAR) ;
> 	{
> 	QString unique_id = QString::fromStdString(meta.mMsgId.toStdString());
> 
>     QList<GxsIdRSTreeWidgetItem*> iteml = groupTreeWidget->findItems(unique_id,Qt::MatchExactly,COL_GROUP_UNIQUE_ID) ;
> 
> if(iteml.empty())
> sitem = new GxsIdRSTreeWidgetItem(item) ;
> else
> sitem = *iteml.begin();
> }


![image](https://user-images.githubusercontent.com/9952056/66684108-88b0d900-ec79-11e9-8e44-188ae87f2143.png)
